src/main/java/com/ot/restaurant/ioc/DependencyContainer.java

```diff
@@ -0,0 +1,54 @@
+package com.ot.restaurant.ioc;
+
+import java.io.File;
+
+public class DependencyContainer<T> {
+  private final IoCContainer ioc = IoCContainer.getInstance();
+
+  private final Object entityType = new Object();
```

Comment:
This is not needed at all. You don't need an entity to start the controllers neither the repositories